sound/soc/codecs/rt5651.c

```diff
@@ -1770,6 +1770,9 @@ static int rt5651_detect_headset(struct snd_soc_component *component)
 
 static bool rt5651_support_button_press(struct rt5651_priv *rt5651)
 {
+	if (!rt5651->hp_jack)
+		return false;
+
 	/* Button press support only works with internal jack-detection */
 	return (rt5651->hp_jack->status & SND_JACK_MICROPHONE) &&
 		rt5651->gpiod_hp_det == NULL;
```

Comment:
@plbossart yes for sure. I guess I'm delaying something with the addition or someone is calling set_jack(NULL) twice. As I understand this will access NULL if you call it twice? Not sure of the semantics of set_jack, I guess you should be able to call it twice with NULL...

Reply:
@juimonen there was a recent addition in soc-core.c to call set_jack(NULL) for robustness. https://github.com/thesofproject/linux/blob/f95e4ebfd5915b02e6e257345b47089ada35bc44/sound/soc/soc-core.c#L935

Do you think this could be the reason?

Reply:
If it is, I think this fix is correct.

Reply:
This is the commit that added this change: 3bb936f5b92a0a6

Reply:
@ranjani yeah could be... I don't have the rt5651 device to test this, usually these devices don't reboot nicely after kernel oops, so it is really hard also with remote access device.